### PR TITLE
feat: add <GasSettingDialog />

### PR DESCRIPTION
**Pull request by UncleBill**

### CODE REVIEW

File: packages/maskbook/tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "rootDir": "./src/",
     "outDir": "./dist/",
+    "baseUrl": ".",
```
Comment (Jack-Works) [inline packages/maskbook/tsconfig.json:6]: Why you're adding this?

**Comment by guanbinrui** [comment]
We also need a toolbar component for opening the dialog. Would you please add it in this PR too?

**Review by guanbinrui — DISMISSED** [review]

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx [hunk 2]
```diff
@@ -40,7 +42,7 @@ export const TransferERC20 = memo<TransferERC20Props>(({ token }) => {
     )
 
     // transfer amount
-    const transferAmount = new BigNumber(amount || '0').multipliedBy(pow10(token.decimals)).toFixed()
+    const transferAmount = formatAmount(amount, token.decimals)
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:45]: Revoke this change.

File: packages/maskbook/tsconfig.json (hunk 1 above)
Comment (UncleBill) [reply packages/maskbook/tsconfig.json]: Removed

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
 } from '@masknet/web3-shared'
 import BigNumber from 'bignumber.js'
-import { TokenAmountPanel } from '@masknet/shared'
+import { TokenAmountPanel, useRemoteControlledDialog } from '@masknet/shared'
+import TuneIcon from '@material-ui/icons/Tune'
 import { SelectTokenDialog } from '../SelectTokenDialog'
 import { useDashboardI18N } from '../../../../locales'
 import { EthereumAddress } from 'wallet.ts'
+import { WalletMessages } from '@masknet/plugin-wallet'
+import { useNativeTokenPrice } from '../../../../../../maskbook/src/plugins/Wallet/hooks/useTokenPrice'
```
Comment (NikoManiac) [inline packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:25]: Don't  direct use maskbook method in dashboard

File: packages/maskbook/src/extension/popups/components/StyledInput/index.tsx [hunk 4]
```diff
@@ -3,18 +3,20 @@ import { TextFieldProps, TextField } from '@material-ui/core'
...
 import { useStylesExtends } from '@masknet/shared'
 
-const useStyles = makeStyles()({
+const useStyles = makeStyles()((theme) => ({
     textField: {
         width: '100%',
     },
     textFieldInput: {
-        backgroundColor: '#F7F9FA',
+        backgroundColor: theme.palette.mode === 'light' ? '#F7F9FA' : theme.palette.background.default,
```
Comment (NikoManiac) [inline packages/maskbook/src/extension/popups/components/StyledInput/index.tsx:11]: Use color constant to replace `#F7F9FA`?

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx (hunk 3 above)
Comment (NikoManiac) [inline packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:25]: Don't direct use maskbook method in dashboard

File: packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx [hunk 5]
```diff
@@ -264,7 +267,7 @@ export function ToolboxHint(props: ToolboxHintProps) {
 
     return (
         <>
-            <div className={classes.wrapper} onClick={openMenu}>
+            <div className={classes.wrapper} onClick={openGasSettingDialog}>
```
Comment (guanbinrui) [inline packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx:270]: Don't forget.

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx (hunk 3 above)
Comment (UncleBill) [reply packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:25]: `theme.palette.background.default` is `#f9fafa`, very close to  `#F7F9FA`, I will use it instead.

File: packages/maskbook/src/extension/popups/components/StyledInput/index.tsx (hunk 4 above)
Comment (UncleBill) [reply packages/maskbook/src/extension/popups/components/StyledInput/index.tsx:11]: `theme.palette.background.default` is `#f9fafa`, very close to  `#F7F9FA`, I will use it instead.

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx (hunk 3 above)
Comment (Jack-Works) [reply packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:25]: Yes

**Review by NikoManiac — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**